src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```diff
@@ -0,0 +1,147 @@
+        return root;
+    }
+
+    private void setRootProperty(FileNodeViewModel fileNode) {
+        root.set(fileNode);
+    }
+
+    public ObservableList<TreeItem<FileNodeViewModel>> getCheckedFileList() {
+        return checkedFileList;
+    }
+
+    public StringProperty texDirectoryProperty() {
+        return texDirectory;
+    }
+
+    public BooleanProperty noFilesFoundProperty() {
+        return noFilesFound;
+    }
+
+    public BooleanProperty searchInProgressProperty() {
+        return searchInProgress;
+    }
+
+    public BooleanProperty successfulSearchProperty() {
+        return successfulSearch;
+    }
+
+    protected void browseButtonClicked() {
+        DirectoryDialogConfiguration directoryDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(initialPath).build();
+
+        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory -> {
+            texDirectory.set(selectedDirectory.toAbsolutePath().toString());
+            preferencesService.setWorkingDir(selectedDirectory.toAbsolutePath());
+        });
+    }
+
+    protected void searchButtonClicked() {
+        WalkFileTreeTask fileTree = new WalkFileTreeTask(getSearchDirectory());
+
+        BackgroundTask.wrap(fileTree::call)
+                      .onRunning(() -> {
+                          noFilesFound.set(true);
+                          searchInProgress.set(true);
+                          successfulSearch.set(false);
+                      })
+                      .onFinished(() -> searchInProgress.set(false))
+                      .onSuccess(root -> {
+                          setRootProperty(root);
+                          noFilesFound.set(false);
+                          successfulSearch.set(true);
+                      })
+                      .onFailure(dialogService::showErrorDialogAndWait)
+                      .executeWith(taskExecutor);
+    }
+
+    private Path getSearchDirectory() {
+        Path directory = Paths.get(texDirectory.get());
+
+        if (Files.notExists(directory)) {
+            directory = initialPath;
+            texDirectory.set(directory.toAbsolutePath().toString());
+        }
+
+        if (!Files.isDirectory(directory)) {
+            directory = directory.getParent();
+            texDirectory.set(directory.toAbsolutePath().toString());
+        }
+
+        return directory;
+    }
+
+    protected void parse() {
+        List<Path> fileList = checkedFileList.stream()
+                                             .map(item -> item.getValue().getFileNode().getPath().toAbsolutePath())
+                                             .filter(path -> path != null && Files.isRegularFile(path))
+                                             .collect(Collectors.toList());
+        if (fileList.isEmpty()) {
+            return;
+        }
+
+        TexParserResult result = new DefaultTexParser().parse(fileList);
```

Comment:
I suspect this needs to run in BackgroundTask as well (since otherwise the UI freezes for a large amount of tex files). Maybe even a small progress indication needs to be added (then preferably in the ParseTexResultView dialog).